util/src/main/java/io/kubernetes/client/util/authenticators/GCPAuthenticator.java

```diff
@@ -37,7 +43,19 @@ public String getToken(Map<String, Object> config) {
 
   @Override
   public boolean isExpired(Map<String, Object> config) {
-    Date expiry = (Date) config.get("expiry");
+    Object expiryObj = config.get("expiry");
+    Date expiry = null;
+    if (expiryObj instanceof Date) {
+      expiry = (Date) expiryObj;
+    } else if (expiryObj instanceof String) {
+      try {
+        expiry = DateFormat.getDateInstance().parse((String) expiryObj);
```

Comment:
[https://github.com/kubernetes/kubernetes/blob/a521285b7cf8273fddf9c8e912070f9e386ffa0d/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/time.go#L151](https://github.com/kubernetes/kubernetes/blob/a521285b7cf8273fddf9c8e912070f9e386ffa0d/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/time.go#L151)

how about explicitly parsing timestamp w/ RFC3339 format here?

Reply:
Turns out there's not a clean way to do that in Java:

https://stackoverflow.com/questions/6038136/how-do-i-parse-rfc-3339-datetimes-with-java/41569330

But I followed the advice to switch to Instant which uses ISO8601 which is nearly identical.